API/Services/Tasks/Scanner/Parser/Parser.cs

```diff
@@ -1106,9 +1107,21 @@ public static float MaxNumberFromRange(string range)
         }
     }
 
+    private static readonly ConcurrentDictionary<string, string> NormalizedCache = 
+        new ConcurrentDictionary<string, string>();
+
     public static string Normalize(string name)
     {
-        return NormalizeRegex.Replace(name, string.Empty).Trim().ToLower();
+        // Check cache first
+        if (NormalizedCache.TryGetValue(name, out string cachedResult))
```

Comment:
Why would we want to cache every string that goes through normalization? That will balloon memory usage.

Reply:
Implementing caching seemed beneficial, given that the Normalize function may be invoked hundreds of times per series, depending on the volume of metadata and chapters. I observed that each series contains a significant amount of recurring metadata. However, I accidently overlooked the necessity of clearing the cache after processing each series.

Reply:
The Normalize is used throughout the codebase, not just the scan loop as well. So this sort of change would not work.